include/seahorn/Transforms/Instrumentation/ShadowMemSeaDsa.hh

```diff
@@ -1,22 +1,51 @@
 /// This file should be completely moved into C++ file.
+// TODO: remove this comment now? we want to use this pass in the vcgen
 #pragma once
 #include "llvm/Pass.h"
 
+#include "sea_dsa/Global.hh"
+#include "sea_dsa/Graph.hh"
+
+#include <memory>
+
 namespace llvm {
 class Value;
 }
+
+namespace {
+// defined in ShadowMemSeaDsa.cc
+class ShadowDsaImpl;
+}
+
 #define USE_NEW_SHADOW_SEA_DSA 1
 namespace seahorn {
+
 class ShadowMemSeaDsa : public llvm::ModulePass {
+
+private:
+  sea_dsa::GlobalAnalysis *m_dsa;
+  std::unique_ptr<ShadowDsaImpl> m_shadow;
+
 public:
   static char ID;
-  ShadowMemSeaDsa() : llvm::ModulePass(ID) {}
+  ShadowMemSeaDsa();
+  ~ShadowMemSeaDsa();
 
   bool runOnModule(llvm::Module &M) override;
 
   bool runOnFunction(llvm::Function &F);
   void getAnalysisUsage(llvm::AnalysisUsage &AU) const override;
   llvm::StringRef getPassName() const override { return "ShadowMemSeaDsa"; }
+
+  // TODO: should this be a cell?
+  bool hasShadowId(const sea_dsa::Node *);
+  unsigned getNodeShadowId(const sea_dsa::Node *);
+
+  sea_dsa::Graph & getSummaryGraph(const llvm::Function &F);
```

Comment:
What is the difference between summary graph and DsaGraph?

Reply:
The summary graph is the bottom up graph. Should I call it directly BUDsaGraph?

Reply:
maybe just a comment on the function will do. I don't have a better name. I prefer current name to `BUDsa`.

Reply:
OK, I will add comments.